unstructured/embed/embedder/open_ai.py

```diff
@@ -0,0 +1,155 @@
+import json
+import os
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Optional
+
+from dataclasses_json import DataClassJsonMixin
+
+from unstructured.documents.elements import (
+    Element,
+)
+from unstructured.ingest.error import EmbedderConnectionError
+from unstructured.ingest.interfaces import (
+    BaseSessionHandle,
+    ConfigSessionHandleMixin,
+    IngestDocSessionHandleMixin,
+)
+from unstructured.ingest.logger import logger
+from unstructured.partition.json import partition_json
+from unstructured.utils import requires_dependencies
+
+
+@requires_dependencies(["langchain"])  # add extras="langchain" when it's added to the makefile
+@dataclass
+class OpenAISessionHandle(BaseSessionHandle):
+    from langchain.embeddings.openai import OpenAIEmbeddings
+
+    service: OpenAIEmbeddings
+
+
+@EmbedderConnectionError.wrap
+@requires_dependencies(["langchain"])  # add extras="langchain" when it's added to the makefile
+def create_openai_object(api_key, model_name):
+    """
+    Creates an OpenAI object to embed elements.
+    Args:
+        api_key: API Key, generated for the user
+        model_name: Model name for OpenAI
+
+    Returns:
+        OpenAI Embeddings object
+    """
+    from langchain.embeddings.openai import OpenAIEmbeddings
+
+    embedder = OpenAIEmbeddings(
```

Comment:
usually don't want to comment on names at this stage but... I'd propose using `encoder` instead of `embedder`. `encoder` is more commonly used.

Reply:
resolving due to duplicate